packages/stores/src/queries-external/position-performance/index.ts

```diff
@@ -0,0 +1,203 @@
+    maxDecimals: 2,
+    locale: "en-US",
+  };
+
+  constructor(
+    kvStore: KVStore,
+    baseURL: string,
+    protected readonly chainGetter: ChainGetter,
+    protected readonly chainId: string,
+    protected readonly priceStore: IPriceStore,
+    protected readonly positionId: string
+  ) {
+    super(kvStore, baseURL, `/cl/v1/position/id/${positionId}`);
+
+    makeObservable(this);
+  }
+
+  protected get chain() {
+    return this.chainGetter.getChain(this.chainId);
+  }
+
+  @computed
+  get feesEarned(): CoinPretty[] {
+    if (!this.response) return [];
+    return (
+      this.response.data?.total_spread_rewards?.map(({ denom, amount }) => {
+        const currency = this.chain.forceFindCurrency(denom);
+        return new CoinPretty(currency, amount);
+      }) ?? []
+    );
+  }
+
+  @computed
+  get incentivesEarned(): CoinPretty[] {
+    if (!this.response) return [];
+    return (
+      this.response.data?.total_incentives_rewards?.map(({ denom, amount }) => {
+        const currency = this.chain.forceFindCurrency(denom);
+        return new CoinPretty(currency, amount);
+      }) ?? []
+    );
+  }
+
+  @computed
+  get totalEarnedValue(): PricePretty {
+    return this.totalEarned.reduce((sum, coin) => {
+      const price = this.priceStore.calculatePrice(coin);
+      if (price) return sum.add(price);
+      else return sum;
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    }, new PricePretty(this.priceStore.getFiatCurrency(this.priceStore.defaultVsCurrency)!, 0));
+  }
+
+  @computed
+  get totalEarned(): CoinPretty[] {
+    // aggregate earned coins by denom
+    const earnedCoinDenomMap = new Map<string, CoinPretty>();
+    [...(this.incentivesEarned ?? []), ...(this.feesEarned ?? [])].forEach(
+      (coin) => {
+        const existingCoin = earnedCoinDenomMap.get(
+          coin.currency.coinMinimalDenom
+        );
+        if (existingCoin) {
+          earnedCoinDenomMap.set(
+            coin.currency.coinMinimalDenom,
+            existingCoin.add(coin)
+          );
+        } else {
+          earnedCoinDenomMap.set(coin.currency.coinMinimalDenom, coin);
+        }
+      }
+    );
+    return Array.from(earnedCoinDenomMap.values());
+  }
+
+  @computed
+  get principal(): { coin: CoinPretty; value: PricePretty }[] {
+    if (!this.response) return [];
+    return (
+      this.response.data?.principal?.assets?.map(({ denom, amount, value }) => {
+        const currency = this.chain.forceFindCurrency(denom);
+        return {
+          coin: new CoinPretty(currency, amount),
+          value: new PricePretty(this.fiatCurrency, value),
+        };
+      }) ?? []
+    );
+  }
+
+  @computed
+  get totalPrincipalValue(): PricePretty {
+    if (!this.response) return new PricePretty(this.fiatCurrency, 0);
+    return new PricePretty(
+      this.fiatCurrency,
+      this.response.data?.principal?.value ?? 0
+    );
+  }
+
+  readonly calculateReturnOnInvestment = computedFn(
```

Comment:
seems like there is some similar logic here to whats in `totalEarned`. Can we make it more dry?